// office-ui-fabric-react/lib-amd/components/ComboBox/ComboBox.doc.js
define(["require", "exports", "react", "./examples/ComboBox.Basic.Example", "./examples/ComboBox.CustomStyled.Example", "./ComboBox.checklist"], function (require, exports, React, ComboBox_Basic_Example_1, ComboBox_CustomStyled_Example_1, ComboBox_checklist_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var ComboBoxBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/examples/ComboBox.Basic.Example.tsx');
    var ComboBoxBasicExampleCodepen = require('!raw-loader!office-ui-fabric-react/lib/codepen/components/ComboBox/ComboBox.Basic.Example.Codepen.txt');
    var ComboBoxCustomStyledExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/examples/ComboBox.CustomStyled.Example.tsx');
    exports.ComboBoxPageProps = {
        title: 'ComboBox',
        componentName: 'ComboBox',
        componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/ComboBox',
        componentStatus: ComboBox_checklist_1.ComboBoxStatus,
        examples: [
            {
                title: 'ComboBox',
                code: ComboBoxBasicExampleCode,
                view: React.createElement(ComboBox_Basic_Example_1.ComboBoxBasicExample, null),
                codepenJS: ComboBoxBasicExampleCodepen
            },
            {
                title: 'ComboBoxCustomStyled',
                code: ComboBoxCustomStyledExampleCode,
                view: React.createElement(ComboBox_CustomStyled_Example_1.ComboBoxCustomStyledExample, null)
            }
        ],
        propertiesTablesSources: [require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/ComboBox.types.ts')],
        overview: require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/docs/ComboBoxOverview.md'),
        bestPractices: '',
        dos: require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/docs/ComboBoxDos.md'),
        donts: require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/docs/ComboBoxDonts.md'),
        isHeaderVisible: true,
        isFeedbackVisible: true,
        allowNativeProps: true
    };
});
