// office-ui-fabric-react/lib/components/ComboBox/ComboBox.doc.js
import * as React from 'react';
import { ComboBoxBasicExample } from './examples/ComboBox.Basic.Example';
import { ComboBoxCustomStyledExample } from './examples/ComboBox.CustomStyled.Example';
import { ComboBoxStatus } from './ComboBox.checklist';
var ComboBoxBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/examples/ComboBox.Basic.Example.tsx');
var ComboBoxBasicExampleCodepen = require('!raw-loader!office-ui-fabric-react/lib/codepen/components/ComboBox/ComboBox.Basic.Example.Codepen.txt');
var ComboBoxCustomStyledExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/examples/ComboBox.CustomStyled.Example.tsx');
export var ComboBoxPageProps = {
    title: 'ComboBox',
    componentName: 'ComboBox',
    componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/ComboBox',
    componentStatus: ComboBoxStatus,
    examples: [
        {
            title: 'ComboBox',
            code: ComboBoxBasicExampleCode,
            view: React.createElement(ComboBoxBasicExample, null),
            codepenJS: ComboBoxBasicExampleCodepen
        },
        {
            title: 'ComboBoxCustomStyled',
            code: ComboBoxCustomStyledExampleCode,
            view: React.createElement(ComboBoxCustomStyledExample, null)
        }
    ],
    propertiesTablesSources: [require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/ComboBox.types.ts')],
    overview: require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/docs/ComboBoxOverview.md'),
    bestPractices: '',
    dos: require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/docs/ComboBoxDos.md'),
    donts: require('!raw-loader!office-ui-fabric-react/src/components/ComboBox/docs/ComboBoxDonts.md'),
    isHeaderVisible: true,
    isFeedbackVisible: true,
    allowNativeProps: true
};
