// office-ui-fabric-react/lib/components/HoverCard/examples/HoverCard.PlainCard.Example.js
import * as tslib_1 from "tslib";
import * as React from 'react';
import { BaseComponent } from 'office-ui-fabric-react/lib/Utilities';
import { HoverCard, HoverCardType } from 'office-ui-fabric-react/lib/HoverCard';
import { DetailsList, buildColumns } from 'office-ui-fabric-react/lib/DetailsList';
import { createListItems } from 'office-ui-fabric-react/lib/utilities/exampleData';
import './HoverCard.Example.scss';
import { Image, ImageFit } from 'office-ui-fabric-react/lib/Image';
import { getColorFromString } from 'office-ui-fabric-react/lib/Color';
var _items;
var HoverCardPlainCardExample = /** @class */ (function (_super) {
    tslib_1.__extends(HoverCardPlainCardExample, _super);
    function HoverCardPlainCardExample(props) {
        var _this = _super.call(this, props) || this;
        _this._onRenderItemColumn = function (item, index, column) {
            var plainCardProps = {
                onRenderPlainCard: _this._onRenderPlainCard,
                renderData: item
            };
            if (column.key === 'color') {
                return (React.createElement(HoverCard, { id: "myID1", plainCardProps: plainCardProps, instantOpenOnClick: true, type: HoverCardType.plain },
                    React.createElement("div", { className: "HoverCard-item", style: { color: item.color } }, item.color)));
            }
            return item[column.key];
        };
        _this._onRenderPlainCard = function (item) {
            var src = item.thumbnail + ("/" + getColorFromString(item.color).hex);
            return React.createElement(Image, { src: src, width: item.width, height: item.height, imageFit: ImageFit.cover });
        };
        _items = _items || createListItems(10);
        _this.state = {
            items: _items,
            columns: _buildColumns()
        };
        return _this;
    }
    HoverCardPlainCardExample.prototype.render = function () {
        var _a = this.state, items = _a.items, columns = _a.columns;
        return (React.createElement("div", null,
            React.createElement("p", null,
                "Hover over the ",
                React.createElement("i", null, "color"),
                " cell of a row item to see the card."),
            React.createElement(DetailsList, { setKey: "hoverSet", items: items, columns: columns, onRenderItemColumn: this._onRenderItemColumn })));
    };
    return HoverCardPlainCardExample;
}(BaseComponent));
export { HoverCardPlainCardExample };
function _buildColumns() {
    return buildColumns(_items).filter(function (column) { return column.name === 'color' || column.name === 'width' || column.name === 'height'; });
}
