// office-ui-fabric-react/lib-amd/components/GroupedList/GroupedList.styles.js
define(["require", "exports", "../../Styling"], function (require, exports, Styling_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var GlobalClassNames = {
        root: 'ms-GroupedList',
        group: 'ms-GroupedList-group',
        link: 'ms-Link',
        listCell: 'ms-List-cell'
    };
    var beziers = {
        easeInOutSine: 'cubic-bezier(0.445, 0.050, 0.550, 0.950)'
    };
    exports.getStyles = function (props) {
        var theme = props.theme, className = props.className;
        var palette = theme.palette;
        var classNames = Styling_1.getGlobalClassNames(GlobalClassNames, theme);
        return {
            root: [
                classNames.root,
                theme.fonts.medium,
                {
                    position: 'relative',
                    fontSize: Styling_1.FontSizes.small,
                    selectors: (_a = {},
                        _a["& :global(." + classNames.listCell + ")"] = {
                            minHeight: 36
                        },
                        _a)
                },
                className
            ],
            group: [
                classNames.group,
                {
                    transition: "background-color " + Styling_1.AnimationVariables.durationValue2 + " " + beziers.easeInOutSine
                }
            ],
            groupIsDropping: [
                {
                    backgroundColor: palette.neutralLight
                }
            ]
        };
        var _a;
    };
});
