// @uifabric/styling/lib/styles/DefaultTypography.js
import { DefaultFontStyles } from './DefaultFontStyles';
import { FontSizes, FontWeights } from './fonts';
export var DefaultFontSizes = {
    mini: FontSizes.mini,
    xSmall: FontSizes.xSmall,
    small: FontSizes.small,
    smallPlus: FontSizes.smallPlus,
    medium: FontSizes.medium,
    mediumPlus: FontSizes.mediumPlus,
    large: FontSizes.large,
    xLarge: FontSizes.xLarge,
    xxLarge: FontSizes.xxLarge,
    mega: FontSizes.mega
};
export var DefaultFontFamilies = {
    default: DefaultFontStyles.medium.fontFamily,
    monospace: 'Menlo, Monaco, "Courier New", monospace'
};
export var DefaultFontWeights = {
    default: FontWeights.regular,
    regular: FontWeights.regular,
    light: FontWeights.light,
    semibold: FontWeights.semibold,
    bold: FontWeights.bold
};
export var DefaultFontVariants = {
    default: {
        family: 'default',
        size: 'inherit'
    },
    caption: {
        size: 'xSmall'
    },
    h1: {
        size: 'mega',
        weight: 'light'
    },
    h2: {
        size: 'xxLarge',
        weight: 'light'
    },
    h3: {
        size: 'xLarge',
        weight: 'light'
    },
    h4: {
        size: 'large',
        weight: 'light'
    },
    h5: {
        size: 'mediumPlus',
        weight: 'light'
    },
    link: {
        color: 'link',
        hoverColor: 'linkHovered'
    }
};
export var DefaultTypography = {
    families: DefaultFontFamilies,
    sizes: DefaultFontSizes,
    weights: DefaultFontWeights,
    variants: DefaultFontVariants
};
