// office-ui-fabric-react/lib-amd/components/HoverCard/examples/HoverCard.PlainCard.Example.js
define(["require", "exports", "tslib", "react", "office-ui-fabric-react/lib/Utilities", "office-ui-fabric-react/lib/HoverCard", "office-ui-fabric-react/lib/DetailsList", "office-ui-fabric-react/lib/utilities/exampleData", "office-ui-fabric-react/lib/Image", "office-ui-fabric-react/lib/Color", "./HoverCard.Example.scss"], function (require, exports, tslib_1, React, Utilities_1, HoverCard_1, DetailsList_1, exampleData_1, Image_1, Color_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var _items;
    var HoverCardPlainCardExample = /** @class */ (function (_super) {
        tslib_1.__extends(HoverCardPlainCardExample, _super);
        function HoverCardPlainCardExample(props) {
            var _this = _super.call(this, props) || this;
            _this._onRenderItemColumn = function (item, index, column) {
                var plainCardProps = {
                    onRenderPlainCard: _this._onRenderPlainCard,
                    renderData: item
                };
                if (column.key === 'color') {
                    return (React.createElement(HoverCard_1.HoverCard, { id: "myID1", plainCardProps: plainCardProps, instantOpenOnClick: true, type: HoverCard_1.HoverCardType.plain },
                        React.createElement("div", { className: "HoverCard-item", style: { color: item.color } }, item.color)));
                }
                return item[column.key];
            };
            _this._onRenderPlainCard = function (item) {
                var src = item.thumbnail + ("/" + Color_1.getColorFromString(item.color).hex);
                return React.createElement(Image_1.Image, { src: src, width: item.width, height: item.height, imageFit: Image_1.ImageFit.cover });
            };
            _items = _items || exampleData_1.createListItems(10);
            _this.state = {
                items: _items,
                columns: _buildColumns()
            };
            return _this;
        }
        HoverCardPlainCardExample.prototype.render = function () {
            var _a = this.state, items = _a.items, columns = _a.columns;
            return (React.createElement("div", null,
                React.createElement("p", null,
                    "Hover over the ",
                    React.createElement("i", null, "color"),
                    " cell of a row item to see the card."),
                React.createElement(DetailsList_1.DetailsList, { setKey: "hoverSet", items: items, columns: columns, onRenderItemColumn: this._onRenderItemColumn })));
        };
        return HoverCardPlainCardExample;
    }(Utilities_1.BaseComponent));
    exports.HoverCardPlainCardExample = HoverCardPlainCardExample;
    function _buildColumns() {
        return DetailsList_1.buildColumns(_items).filter(function (column) { return column.name === 'color' || column.name === 'width' || column.name === 'height'; });
    }
});
